tests/dags/providers/provider_api_scripts/test_museum_victoria.py

```diff
@@ -1,182 +1,200 @@
 
-    with patch.object(mv.delay_request, "get", return_value=r) as mock_call:
-        actual_response = mv._get_batch_objects(query_param)
-
-    assert actual_response is None
-    assert mock_call.call_count == 3
-
-
-def test_get_media_info_success():
+def test_get_images_success():
     media = _get_resource_json("media_data_success.json")
-    actual_image_data = mv._get_media_info(media)
-
-    expected_image_data = _get_resource_json("image_data_success.json")
+    actual_image_data = mv._get_images([media])
+
+    expected_image_data = {
+        "creator": "Photographer: Deb Tout-Smith",
+        "foreign_identifier": "media/329745",
+        "image_url": "https://collections.museumsvictoria.com.au/content/media/45/329745-large.jpg",
+        "license_info": get_license_info(
+            license_url="https://creativecommons.org/licenses/by/4.0"
+        ),
+        "width": 2785,
+        "height": 2581,
+    }
 
-    assert actual_image_data == expected_image_data
+    assert actual_image_data[0] == expected_image_data
 
 
 def test_get_media_info_failure():
     media = _get_resource_json("media_data_failure.json")
-    actual_image_data = mv._get_media_info(media)
+    actual_image_data = mv.get_record_data(media)
 
-    assert len(actual_image_data) == 0
+    assert actual_image_data is None
 
 
 def test_get_image_data_large():
     image_data = _get_resource_json("large_image_data.json")
 
-    actual_image_url, actual_height, actual_width = mv._get_image_data(image_data)
+    actual_image_url, actual_height, actual_width, actual_filesize = mv._get_image_data(
+        image_data
+    )
 
     assert actual_image_url == (
         "https://collections.museumsvictoria.com.au/content/media/45/"
         "329745-large.jpg"
     )
     assert actual_height == 2581
     assert actual_width == 2785
+    assert actual_filesize == 890933
 
 
 def test_get_image_data_medium():
     image_data = _get_resource_json("medium_image_data.json")
 
-    actual_image_url, actual_height, actual_width = mv._get_image_data(image_data)
+    actual_image_url, actual_height, actual_width, actual_filesize = mv._get_image_data(
+        image_data
+    )
 
     assert actual_image_url == (
         "https://collections.museumsvictoria.com.au/content/media/45/"
         "329745-medium.jpg"
     )
     assert actual_height == 1390
     assert actual_width == 1500
+    assert actual_filesize == 170943
 
 
 def test_get_image_data_small():
     image_data = _get_resource_json("small_image_data.json")
 
-    actual_image_url, actual_height, actual_width = mv._get_image_data(image_data)
+    actual_image_url, actual_height, actual_width, actual_filesize = mv._get_image_data(
+        image_data
+    )
 
     assert actual_image_url == (
         "https://collections.museumsvictoria.com.au/content/media/45/"
         "329745-small.jpg"
     )
     assert actual_height == 500
     assert actual_width == 540
+    assert actual_filesize == 20109
```

Comment:
Should we be checking `filetype` in any of these tests as well?

Reply:
I've added a test for filetype that mocks image_store's `save_item` method. It passes if I run that test alone, but fails when I run all the tests together (and in CI). I need help :)